src/main/kotlin/net/lumalyte/lg/infrastructure/services/BankInterestScheduler.kt

```
@@ -0,0 +1,62 @@
+    /** Starts the periodic scheduler. Safe to call once at plugin enable. */
+    fun start() {
+        if (scheduledTask != null) return
+
+        scheduledTask = object : BukkitRunnable() {
+            override fun run() {
+                try {
+                    val credited = bankAutomationService.accrueInterest()
+                    if (credited > 0) {
+                        logger.info("Bank interest accrued for $credited guild(s)")
+                    }
+                } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)